pkg/config/types.go

```diff
@@ -58,6 +60,8 @@ type UserMapping struct {
 
 // Config specifies the configuration for a aws-iam-authenticator server
 type Config struct {
```

Comment:
The config here is used for the startup config file.  A server should be created with a config but we shouldn't be modifying it.  I think the canonicalized map and the lock should be wrapped into another structure that the server uses and passes to the handler.